apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs

```diff
@@ -333,12 +307,35 @@ public override int GetHashCode()
         {
             unchecked
             {
-                int hashCode = StructMembers?.Count ?? 0;
+                int hashCode = Lists.GetHashCode(StructFields);
                 hashCode = (hashCode * 397) ^ Size.GetValueOrDefault(0);
                 hashCode = (hashCode * 397) ^ (int)TypeCode;
                 hashCode = (hashCode * 397) ^ (ArrayElementType?.GetHashCode() ?? 0);
                 return hashCode;
             }
         }
+
+        /// <summary>
+        /// Value type representing the name and type of a field within a struct. This is like SpannerStruct.Field
+        /// but without the value - and it's private. This mostly makes equality comparisons simpler.
+        /// </summary>
+        private struct StructField : IEquatable<StructField>
+        {
+            internal string Name { get; }
+            internal SpannerDbType Type { get; }
+
+            public StructField(string name, SpannerDbType type)
+            {
+                Name = GaxPreconditions.CheckNotNull(name, nameof(name));
+                Type = GaxPreconditions.CheckNotNull(type, nameof(type));
+            }
+
+            public override bool Equals(object obj) => obj is StructField sf ? Equals(sf) : false;
```

Comment:
What about `obj is StructField sf && Equals(sf);`?

Reply:
Done. (Not sure why I didn't think of that...)